### ISSUE TITLE
ESM Support

### ISSUE THREAD

**Issue by Bykiev** [Stuk/jszip#867]
Hi, is any plans to add ESM support? All modern libraries already support it

**Comment by jonkoops** [comment]
Yes, this is planned. It will likely be part of a next major version.

**Comment by jogibear9988** [comment]
Is there a way to help to get a ESM version? I would love not to create a fork.
I work on complete bundle free apps, and so I need ESM code.

**Comment by jogibear9988** [comment]
What about using Typescript?

**Comment by jogibear9988** [comment]
I've created a pull: #960 
I'll upload also a own package if this will not be merged

**Comment by ChristianLutz** [comment]
@jogibear9988 See also https://github.com/telerik/jszip-esm

**Comment by jogibear9988** [comment]
I saw this, but only after I was finished...
and mine is based on the newer version

**Comment by hlovdal** [comment]
This library has turned into a liability and has started breaking the build because it does not support ESM:

```text
Warning: ...\src\app\shared\zipfile\zipfile.service.ts depends on 'jszip'. CommonJS or AMD dependencies can cause optimization bailouts.
For more info see: https://angular.io/guide/build#configuring-commonjs-dependencies

./src/app/shared/zipfile/zipfile.service.ts:1:0-188 - Error: Module not found: Error: Can't resolve '.../node_modules/@angular-devkit/build-angular/node_modules/@babel/runtime/helpers/esm/asyncToGenerator.js' in '...\src\app\shared\zipfile'
```

Please update, the time is way overdue to switch to ESM.

**Comment by bybraveHQ** [comment]
For those still waiting on this: since jszip hasn't had a release since 2022, I published a maintained fork with ESM support shipped: **[@bybrave/jszip2](https://github.com/bybraveHQ/jszip2)** (`npm i @bybrave/jszip2`).

```js
import JSZip from "@bybrave/jszip2";
```

- Proper `exports` map (`import` / `require` / `types` conditions) — works in Node ESM, TypeScript `node16`/`bundler` resolution, and modern bundlers without config.
- The API is unchanged — drop-in replacement for `jszip@3.10.1`.
- The runtime is modernized along the way: native Promises (no `lie`), no `setimmediate` polyfill (fixes the sandboxed-environment crashes from #909/#934), pako 2, and no `readable-stream` (no more `Cannot find module 'stream'` with webpack 5).

Dependency count went from 4 to 1 (`pako`). Tests are the original qunit suite (189 green on Node 18/20/22) plus bundle smoke tests.

**Comment by jogibear9988** [comment]
@bybraveHQ 

i already created a fork of it:

https://www.npmjs.com/package/@node-projects/jszip
https://github.com/node-projects/jszip
